### PR TITLE
apps/golang: fix flush and enable span sampling tests

### PR DESCRIPTION
**Pull request by katiehockman**
## Description
Enable single span sampling tests for the Go Tracer after fixes in https://github.com/DataDog/dd-trace-go/pull/1509.

Some bugs in the test setup needed to be fixed in order to accomplish this:
  - The Flush APIs needed to call Flush instead of Stop in the single span tests. For the tracestats tests, they needed to call Stop, so a new API endpoint was added to allow the tracestats to stop the tracer explicitly. See https://github.com/DataDog/system-tests/issues/596 for more details.
  - One of the tests was flaky at 1 second. Moving it to 2 appears to deflake this.

This PR updates the go.mod to point to https://github.com/DataDog/dd-trace-go/pull/1509, which is now merged.

## Check list

Your PR is not ready to be reviewed? Please save it as a draft :pray:

- [ ] Follows the style guidelines of this project (See [how to easily lint the code](https://github.com/DataDog/system-tests/blob/main/docs/edit/lint.md))
- [ ] CI is passing
- [ ] There is at least one approval from code owners

Yes to all? Feel free to merge it whenever you want :heart:

### CODE REVIEW

File: parametric/test_span_sampling.py [hunk 1]
```diff
@@ -204,15 +196,14 @@ def test_single_rule_rate_limiter_span_sampling_sss008(test_agent, test_library)
     assert_sampling_decision_tags(span, sample_rate=None, mechanism=None)
 
     # wait a second for rate limiter tokens to replenish
-    time.sleep(1)
+    time.sleep(2)
```
Comment (Kyle-Verhoog) [inline parametric/test_span_sampling.py:199]: is this change intentional?

**Review by Kyle-Verhoog — APPROVED** [review]

File: parametric/test_span_sampling.py (hunk 1 above)
Comment (ZStriker19) [reply parametric/test_span_sampling.py:199]: Seems like yes: https://github.com/DataDog/system-tests/pull/576#issue-1411467823

**Review by ZStriker19 — APPROVED** [review]

File: parametric/test_span_sampling.py (hunk 1 above)
Comment (katiehockman) [reply parametric/test_span_sampling.py:199]: Yep. When I was testing it locally it was a bit flaky. It was giving 60% when it wanted 100%, probably because the sampling was done over a time period that may have overlapped here. Giving it an extra second for the rate limiter to catch up fixed the issue.

File: parametric/test_library_tracestats.py [hunk 2]
```diff
@@ -255,7 +268,7 @@ def test_top_level_TS005(library_env, test_agent, test_library):
 
 @pytest.mark.skip_library("nodejs", "nodejs has not implemented stats computation yet")
 @enable_tracestats()
-def test_successes_errors_recorded_separately_TS006(library_env, test_agent, test_library):
+def test_successes_errors_recorded_separately_TS005(library_env, test_agent, test_library, test_server):
```
Comment (Kyle-Verhoog) [inline parametric/test_library_tracestats.py:271]: Can we revert the changes to the test names? They're supposed to line up with the names here: https://docs.google.com/document/d/159HICPGmFQW1mW6sFHyNL0GKYPExQ6cWNHY31jec3mc/edit#heading=h.all329scnpmr

**Review by Kyle-Verhoog — CHANGES_REQUESTED** [review]

File: parametric/test_library_tracestats.py (hunk 2 above)
Comment (katiehockman) [reply parametric/test_library_tracestats.py]: Can do 👍

**Review by Kyle-Verhoog — APPROVED** [review]
🙌

**Review by cbeauchesne — APPROVED** [review]
